R/parameters_selection.R

```diff
@@ -0,0 +1,141 @@
+#' Parameters Selection
+#'
+#' Automated selection of the 'best' parameters. For frequentist simple GLMs, it performes an AIC-based stepwise selection. For Bayesian models, it uses the \code{projpred} package.
+#'
+#' @param model A statistical model.
+#' @param ... Arguments passed to or from other methods.
+#'
+#'
+#' @examples
+#' model <- lm(mpg ~ ., data = mtcars)
+#' parameters_selection(model)
+#'
+#' model <- lm(mpg ~ cyl * disp * hp * wt , data = mtcars)
+#' parameters_selection(model, cross_validation = FALSE)
+#'
+#' \dontrun{
+#' # lme4 -------------------------------------------
+#' library(lme4)
+#' model <- lmer(Sepal.Width ~ Sepal.Length * Petal.Width * Petal.Length + (1|Species), data = iris)
+#' parameters_selection(model)
+#'
+#' # rstanarm -------------------------------------------
+#' library(rstanarm)
+#' model <- stan_glm(mpg ~ ., data = mtcars)
+#' parameters_selection(model, cross_validation = TRUE)
+#'
+#' model <- stan_glm(mpg ~ cyl * disp * hp * wt , data = mtcars)
+#' parameters_selection(model, cross_validation = FALSE)
+#' }
+#'
+#'
+#' @export
+parameters_selection <- function(model, ...){
+  UseMethod("parameters_selection")
+}
+
+
+
+#' @rdname parameters_selection
+#' @importFrom stats step
+#' @export
+parameters_selection.lm <- function(model, ...){
+  junk <- capture.output(best <- step(model, ...))
+
+  parameters <- names(best$coefficients)
+  formula <- .reconstruct_formula(parameters)
+  formula
+}
+
+
+
+
+
+
+
+
+
+#' @rdname parameters_selection
+#' @export
+parameters_selection.merMod <- function(model, ...){
+  if (!requireNamespace("cAIC4", quietly = TRUE)) {
+    stop("Package `cAIC4` required. Please install it by running `install.packages(cAIC4)`.", call. = FALSE)
+  }
+
+  best <- cAIC4::stepcAIC(model)
```

Comment:
This could go into performance.

Reply:
Since it's really about parameters selection (rather than assessing or comparing AICs per se)  I feel like it fits here

Reply:
Ok, but maybe we could think of adding the cAIC4 to performance?

Reply:
Sure, although I don't know what it's worth outside of variable selection, I haven't read the paper very thoroughly 😅